docker/generate_makefile.py

```diff
@@ -4,200 +4,141 @@
-    fuzzers/{fuzzer}
-
-.pull-{fuzzer}-{benchmark}-intermediate-runner: pull-base-runner
-	docker pull {base_tag}/runners/{fuzzer}/{benchmark}-intermediate
-
-.{fuzzer}-{benchmark}-runner: .{fuzzer}-{benchmark}-builder .{fuzzer}-{benchmark}-intermediate-runner
-	docker build \\
-    --tag {base_tag}/runners/{fuzzer}/{benchmark} \\
-    --build-arg fuzzer={fuzzer} \\
-    --build-arg benchmark={benchmark} \\
-    $(call cache_from,{base_tag}/runners/{fuzzer}/{benchmark}) \\
-    --file docker/benchmark-runner/Dockerfile \\
-    .
-
-.pull-{fuzzer}-{benchmark}-runner: .pull-{fuzzer}-{benchmark}-builder .pull-{fuzzer}-{benchmark}-intermediate-runner
-	docker pull {base_tag}/runners/{fuzzer}/{benchmark}
-
-""" + FUZZER_BENCHMARK_RUN_TARGETS_TEMPLATE + """
-
-else
-
-build-{fuzzer}-{benchmark}: .{fuzzer}-{benchmark}-builder
-pull-{fuzzer}-{benchmark}: .pull-{fuzzer}-{benchmark}-builder
-
-endif
-"""
-
-
-def generate_fuzzer(fuzzer, benchmarks):
-    """Output make rules for a single fuzzer."""
-    # Generate build rules for the fuzzer itself.
-    print(FUZZER_TEMPLATE.format(fuzzer=fuzzer, base_tag=BASE_TAG))
-
-    # Generate rules for fuzzer-benchmark pairs.
-    for benchmark in benchmarks:
-        print(
-            FUZZER_BENCHMARK_TEMPLATE.format(fuzzer=fuzzer,
-                                             benchmark=benchmark,
-                                             base_tag=BASE_TAG))
+from common import yaml_utils
+from common import benchmark_utils
+from common import fuzzer_utils
+from experiment.build import docker_images
 
-    # Generate rules for building/pulling all target/benchmark pairs.
-    all_build_targets = ' '.join(
-        ['build-{0}-{1}'.format(fuzzer, benchmark) for benchmark in benchmarks])
-    all_pull_targets = ' '.join(
-        ['pull-{0}-{1}'.format(fuzzer, benchmark) for benchmark in benchmarks])
-    print('build-{fuzzer}-all: {all_targets}'.format(
-        fuzzer=fuzzer, all_targets=all_build_targets))
-    print('pull-{fuzzer}-all: {all_targets}'.format(
-        fuzzer=fuzzer, all_targets=all_pull_targets))
+BASE_TAG = "gcr.io/fuzzbench"
+BENCHMARK_DIR = benchmark_utils.BENCHMARKS_DIR
+
+
+def _print_benchmark_fuzz_target(benchmarks):
+    """Prints benchmark variables from benchmark.yaml files."""
+    for benchmark in benchmarks:
+        benchmark_vars = yaml_utils.read(
+            os.path.join(BENCHMARK_DIR, benchmark, 'benchmark.yaml'))
+        print(benchmark + '-fuzz-target=' + benchmark_vars['fuzz_target'])
+        print()
+
+
+def _print_makefile_run_template(image):
+    fuzzer, benchmark = image['tag'].split('/')[1:]
+
+    for run_type in ('run', 'debug', 'test-run'):
+        print(('{run_type}-{fuzzer}-{benchmark}: ' +
+               '.{fuzzer}-{benchmark}-runner').format(run_type=run_type,
+                                                      benchmark=benchmark,
+                                                      fuzzer=fuzzer))
+
+        print("""\
```

Comment:
single quotes for strings except docstrings.

Reply:
Done